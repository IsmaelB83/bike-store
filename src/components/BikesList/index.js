import { connect } from 'react-redux';

import BikesList from './BikesList';

function mapStateToProps(state) {
  return {
    bikes: state.bikes,
  };
}

export default connect(mapStateToProps)(BikesList);
