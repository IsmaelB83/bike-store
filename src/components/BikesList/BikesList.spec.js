import React from 'react';
import { shallow } from 'enzyme';

import BikesList from './BikesList';

describe('BikesList', () => {
  const defaultProps = {
    addToCart: jest.fn(),
    bikes: [
      {
        id: 'bike1',
        name: 'mountain cf 7.0',
        image: 'url/to/image',
        price: 3999.99,
        hasStock: true,
      },
    ],
    className: 'test',
  };
  const render = props => shallow(<BikesList {...defaultProps} {...props} />);
  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list of bikes', () => {
    expect(wrapper.find('.list').props().items).toHaveLength(1);
  });

  it('should add a bike', () => {
    const bike = defaultProps.bikes[0];
    const list = wrapper
      .find('.list')
      .props()
      .renderItem(bike);
    const { onAddToCartClick } = list.props;
    onAddToCartClick();
    expect(defaultProps.addToCart).toHaveBeenCalledWith(bike.id);
  });
});
