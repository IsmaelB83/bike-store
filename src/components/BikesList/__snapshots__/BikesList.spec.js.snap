// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikesList should render 1`] = `
<div
  className="bikes-list test"
>
  Bikes List
  <List
    className="list"
    items={
      Array [
        Object {
          "hasStock": true,
          "id": "bike1",
          "image": "url/to/image",
          "name": "mountain cf 7.0",
          "price": 3999.99,
        },
      ]
    }
    renderItem={[Function]}
  />
</div>
`;
